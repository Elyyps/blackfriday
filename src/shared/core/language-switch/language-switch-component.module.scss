.menu-language-switch {
  display: flex;
  flex-direction: row;
  div {
    margin-left: 8px;
    margin-right: 8px;
    font-size: 14px;
    color: white;
  }
  &-option {
    color: $brand-primary !important;
  }
  span {
    height: 25px;
    font-family: $global-font-family-secondary;
    font-size: 14px;
    font-weight: 600;
    text-decoration: none;
    color: #fff;
    &:hover {
      cursor: pointer;
      color: $global-link-color !important;
    }
  }
}
