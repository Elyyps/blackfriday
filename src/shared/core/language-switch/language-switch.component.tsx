import * as React from "react";
import styles from "./language-switch-component.module.scss";
import { ILanguageSwitchContainerProps } from "./language-switch.container";
import { LOCALES } from "@app/constants";
import { InjectedIntlProps, injectIntl } from "react-intl";

export interface ILanguageSwitchComponentProps {}

const component = (props: ILanguageSwitchContainerProps & InjectedIntlProps) => (
  <div className={styles["menu-language-switch"]}>
    <span
      role="button"
      onClick={() => {
        props.onLanguageSwitch(LOCALES.NL);
      }}
      style={props.currentLocale === LOCALES.NL ? { color: "#e10a14" } : { color: "#ffffff" }}
    >
      Nederlands
    </span>
    <div>/</div>
    <span
      role="button"
      onClick={() => {
        props.onLanguageSwitch(LOCALES.FR);
      }}
      style={props.currentLocale === LOCALES.FR ? { color: "#e10a14" } : { color: "#ffffff" }}
    >
      Fran√ßais
    </span>
  </div>
);
const LanguageSwitchComponent = injectIntl(component);
export { LanguageSwitchComponent };
