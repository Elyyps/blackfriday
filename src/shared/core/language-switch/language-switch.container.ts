import { connect, MapStateToProps, MapDispatchToProps } from "react-redux";
import { LanguageSwitchComponent } from "./language-switch.component";
import { IAppState } from "@app/stores";
import { localesActions } from "@app/stores/locales";
import { IBlackFridayRootURL } from "@app/stores/settings";

interface IStateProps {
  blackFridayRootUrl: IBlackFridayRootURL | undefined;
  currentLocale: string;
}
interface IDispatchProps {
  onLanguageSwitch: (locale: string) => void;
}

const mapStateToProps: MapStateToProps<IStateProps, {}, IAppState> = state => {
  const { locale } = state.locales;
  const blackFridayRootUrl = state.settings.blackFridayRootUrl;

  return { currentLocale: locale, blackFridayRootUrl };
};

const mapDispatchToProps: MapDispatchToProps<IDispatchProps, {}> = dispatch => ({
  onLanguageSwitch: (locale: string) => dispatch(localesActions.setLocale({ locale }))
});

export interface ILanguageSwitchContainerProps extends IStateProps, IDispatchProps {}
export const LanguageSwitchContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(LanguageSwitchComponent);
