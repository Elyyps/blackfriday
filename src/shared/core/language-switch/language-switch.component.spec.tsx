import React from "react";
import { shallow } from "enzyme";
import { LanguageSwitchComponent } from "./language-switch.component";

describe("[LanguageSwitch]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <LanguageSwitchComponent currentLocale="nl" onLanguageSwitch={() => jest.fn()} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
