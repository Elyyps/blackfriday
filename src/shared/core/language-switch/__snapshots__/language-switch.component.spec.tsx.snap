// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LanguageSwitch] should render component without crashing 1`] = `
<a
  aria-label="languageSwitchButton"
  className="language-switch"
  onClick={[Function]}
  role="button"
>
  <img
    alt="image"
    src={
      Object {
        "ReactComponent": [Function],
        "__esModule": true,
        "default": "language.svg",
      }
    }
  />
  <span>
    <ResourceTextComponent
      resourceKey="language-switch_en"
    />
  </span>
</a>
`;
